handout/forms/form-builder/form-builder_custom_validation.md

```diff
@@ -0,0 +1,102 @@
+# FormBuilder Custom Validation
+
+As useful as the built-in validators are, it is very useful to be able to include your own.
+Angular 2 allows you to do just that, with minimal effort.
+
+Let's assume we are using the same Login Form, but now we also want to test that our password has an exclamation mark somewhere in it.
+
+_app/login-form.component.ts_
+```ts
+function hasExclamationMark (input: FormControl) {
+  const hasExclamation = input.value.indexOf('!') >= 0;
+
+  return hasExclamation ? null : { needsExclamation: true };
```

Comment:
In this case, I'm fine with the ternary operator. In a case like this it's clear and concise, and in case someone asks "whats that?", its a quick aside conversation.

When I see stuff like

```
return  w ? (x ? 'yes' : 'what ') : y ? 'no' : 'huh'
```

... then I'd raise issues over readability/understandability;